driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java

```diff
@@ -14,48 +14,58 @@
  * limitations under the License.
  */
 
-package com.mongodb.internal.connection;
+package com.mongodb.internal.operation;
 
 import com.mongodb.MongoNamespace;
 import com.mongodb.ServerAddress;
 import com.mongodb.ServerCursor;
 import com.mongodb.lang.Nullable;
+import org.bson.BsonDocument;
+import org.bson.BsonTimestamp;
 
 import java.util.List;
 
+import static com.mongodb.assertions.Assertions.isTrue;
+
 /**
- * A batch of query results.
+ * The command cursor result
  *
  * <p>This class is not part of the public API and may be removed or changed at any time</p>
  */
-public class QueryResult<T> {
-    private final MongoNamespace namespace;
+public class CommandCursorResult<T> {
+
+    private static final String CURSOR = "cursor";
+    private static final String POST_BATCH_RESUME_TOKEN = "postBatchResumeToken";
+    private static final String OPERATION_TIME = "operationTime";
+    private final BsonDocument cursorDocument;
```

Comment:
Intellij points out that this field can be converted to a local variable in the constructor.

Reply:
Done